src/data/markdown/translated-guides/en/02 Using k6/06 Test life cycle.md

```diff
@@ -160,11 +158,8 @@ You might have noticed the function signatures of the `default` and `teardown` f
 take an argument, referred to here as `data`.
 
 This `data` is whatever the `setup` function returns.
-Having a mechanism for passing data from the setup stage to the subsequent VU and teardown stages is compatible with our goal of supporting local, cloud, and clustered execution modes without requiring script changes when switching between them.
-In cloud and clustered modes, the node that runs the setup code might not be the node that runs the teardown code.
 
-To support these modes, you can pass only data (i.e. JSON) between `setup()` and the
-other stages.
+You can pass only data (i.e. JSON) between `setup()` and the other stages.
```

Comment:
Maybe a note that it's a copy of what `setup` returns. So that it:
1. uses more memory if it's big
2. you can't actually change it in the VU code and see the changes in the teardown. Although any VU will see it's changes to the `data`. This though is likely to change as we might freeze the whole object just to make it less obvious that it won't work

Reply:
> Although any VU will see it's changes to the data. This though is likely to change as we might freeze the whole object just to make it less obvious that it won't work

@MStoykov 

Sorry, I'm a bit confused here. So, VU code can manipulate data from setup, but it can't pass that data to teardown. Is that correct?

And, in the table, I note that setup code can be used to " share data among VUs". How is this data shared. Is it from one VU to the next VU? Or does each VU grab "clean" data from the `setup` code?

However it works, it seems like, if VU code can manipulate data that doesn't end up in the teardown stage, then the data is not really "passed" from Setup to VU to Default. 

Rather, setup passes data in two places. Once, directly to `teardown`. And once (or many times??) directly to `default`.

I've made a small diagram to illustrate how I conceive this. Excuse me, I'm sure said some inaccurate things here, but I thought it may be easier to correct than to explain :-) .

![image](https://user-images.githubusercontent.com/47385188/164256189-872d3d56-0c1d-4c2e-9a91-09b6b9fe7d4b.png)

Reply:
more or less -yes to all of those :). 

In practice implementing `data` being modified and all that modifications being visible by :
1. all VUs
2. teardown
Will be ... extremely complicated and (computationally) expensive especially in distributed execution. Which is why we haven't even discussed it ever. But long long time ago I made fixes so that they have separate copies - previous to that this was sometimes panicking as multiple VUs were writing to the same not thread safe structure.

Reply:
Haha, this sounds pretty complicated. I think I'm just going to just suggest that it's a "copy," and leave it at that. Especially since it sounds like this data may get "frozen" in the VU stage in some later k6 version anyway.  

 For the people who are motivated enough to actually use this implementation in some advanced way, well they can look at the source code (and find this discussion). :slightly_smiling_face: